src/main/java/reposense/authorship/model/AuthorshipSummary.java

```diff
@@ -1,36 +1,56 @@
 package reposense.authorship.model;
 
 import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.List;
 
 import reposense.model.Author;
+import reposense.model.Format;
 
 /**
  * Stores the contribution summary of the authors in the repo.
  */
 public class AuthorshipSummary {
     private final List<FileResult> fileResults;
     private final HashMap<Author, Integer> authorFinalContributionMap;
+    private final HashMap<Author, LinkedHashMap<String, Integer>> authorFileTypeContributionMap;
```

Comment:
`HashMap`/`LinkedHashMap` -> `Map`, let's try to use interfaces as much as we can (can change the 1 above as well if you want).